include/zephyr/net/wifi_mgmt.h

```diff
@@ -511,6 +527,10 @@ struct wifi_reg_domain {
 	bool force;
 	/** Country code: ISO/IEC 3166-1 alpha-2 */
 	uint8_t country_code[WIFI_COUNTRY_CODE_LEN];
+	/** Number of channel list information */
+	unsigned int num_channels;
+	/** All channel info*/
+	struct wifi_reg_chn_info chnl_info[MAX_REG_CHNL_NUM];
```

Comment:
This looks quite bad, the array is 462 bytes long which causes lot of stack usage. Would there be any other way to store the information?

Reply:
Changed to use heap memory. 
Protected under flag "CONFIG_HEAP_MEM_POOL_SIZE"  for platforms not using heap.